### DISCUSSION TITLE
How to import local custom component?

### DISCUSSION THREAD

**Discussion by AlexRadch** [vuepress/core#1207] [Q&A]
I created local custom components and want to use them only on one page. When I try to import and use them:
``` md
<script>
    import CustomComponent from './CustomComponent.vue'
</script>

<CustomComponent />
```
I get the next error:
```
[plugin:vite:import-analysis] Failed to resolve import "./CustomComponent.vue" from "src\.vuepress\.temp\pages\HappyNewYear\index.html.vue". Does the file exist?

C:/AProj/AlexRadch/src/.vuepress/.temp/pages/HappyNewYear/index.html.vue:9:32

1  |  
2  |      import CustomComponent from './CustomComponent.vue'
   |                                 ^
3  |  
4  |  const _sfc_main = {}
```

How can I use local custom components only on one page and not register them globally?

As I see VitePress support such registration, but why VuePress does not?

**Comment by Mister-Hope** [comment]
https://vuepress-theme-hope.github.io/v2/cookbook/advanced/component.html

**Reply by AlexRadch** [reply]
Do you suggest creating aliases for each component? ` "@MyComponent": path.resolve(__dirname, "components/MyComponent.vue"),`.
CustomComponent is located in the same folder as the `.md` file located.
Is there a more convenient way to not register aliases in the global config?

Why VitePress support such local registration,  but VuePress does not?

**Reply by AlexRadch** [reply]
>In addition, since Markdown will be converted to Vue single-file components in the cache directory, any relative path import will be invalid in Vue SFC.

Can VuePress add there some support to import files related to the original `md` file folder?
For example to support the `@currentPath` or  `@currentDir` alias or something similar?

For example `import CustomComponent from "@currentPath/MyComponent";` where `@currentPath` alias is already defined by VuePress to `md` file folder?

**Reply by Mister-Hope** [reply]
cc @meteorlxy
